scripts/build.js

```diff
@@ -53,25 +97,33 @@ function afterBuild(options) {
  */
 
 function build(options) {
-  var args = [path.join('node_modules', 'pangyp', 'bin', 'node-gyp'), 'rebuild'].concat(
-    ['libsass_ext', 'libsass_cflags', 'libsass_ldflags', 'libsass_library'].map(function(subject) {
-      return ['--', subject, '=', process.env[subject.toUpperCase()] || ''].join('');
-    })).concat(options.args);
+  installGitDependencies(function(err) {
+    if (err) {
+      console.error(err.message);
+      process.exit(1);
+    }
 
-  console.log(['Building:', process.sass.runtime.execPath].concat(args).join(' '));
+    var args = [path.join('node_modules', 'pangyp', 'bin', 'node-gyp'), 'rebuild'].concat(
+      ['libsass_ext', 'libsass_cflags', 'libsass_ldflags', 'libsass_library'].map(function(subject) {
+        return ['--', subject, '=', process.env[subject.toUpperCase()] || ''].join('');
+      })).concat(options.args);
 
-  var proc = spawn(process.sass.runtime.execPath, args, {
-    stdio: [0, 1, 2]
-  });
+    console.log(['Building:', process.sass.runtime.execPath].concat(args).join(' '));
 
-  proc.on('exit', function(errorCode) {
-    if (!errorCode) {
-      afterBuild(options);
+    var proc = spawn(process.sass.runtime.execPath, args, {
+      stdio: [0, 1, 2]
+    });
 
-      return;
-    }
+    proc.on('exit', function(errorCode) {
+      if (!errorCode) {
+        afterBuild(options);
 
-    console.error(errorCode === 127 ? 'node-gyp not found!' : 'Build failed');
+        return;
+      }
+
+      console.error(errorCode === 127 ? 'node-gyp not found!' : 'Build failed');
+      process.exit(1);
```

Comment:
Currently if the postinstall script is run directly and building fails it exits with a `0`.

```
npm run postinstall
```